shell/store/index.js

```diff
@@ -81,7 +81,13 @@ const getActiveNamespaces = (state, getters) => {
     return out;
   }
 
-  const namespaces = getters[`${ inStore }/all`](NAMESPACE);
+  let namespaces = [];
+
+  if (Array.isArray(state.allNamespaces) && state.allNamespaces.length > 0) {
```

Comment:
I think removing getters from updateNamespaces makes sense but I'm less sure about this change. Is there some benefit to using `allNamespaces` instead of `getters[`${ inStore }/all`](NAMESPACE)` ? It looks like it would work but I don't understand the motivation for the change.

I'm also curious as to why `allNamespaces` exists at all. I see how it is set, but (aside from the change in this PR) I don't see where it is accessed. @catherineluse and @richard-cox have more experience with namespace filtering, maybe they can clarify.

Reply:
`allNamespace` can be controlled by different products. For example, harvester does not need to display the system namespace.

If we use `getters[' ${inStore}/all '](NAMESPACE)`, then when the user clicks `All Namespaces`, `getters.isAllNamespaces` is always `true`, All namespaces (including the same namespace) will be displayed。

Reply:
Okay, thank you for explaining. I think this change is ok then